### PR TITLE
Optimize audio node rendering

### PR DESCRIPTION
**Pull request by mzgoddard**
This improves node rendering performance by reusing PIXI objects that represent how they are rendered in WebGL.

There are two types of objects cached with this change:
- PIXI.Text are cached as textures that are reused as PIXI.Sprites
- PIXI.Graphics representing the node's rectangle and unconnected ports create a PIXI.GraphicsGeometry that can be reused by other PIXI.Graphics objects

Here are the types of work this change reduces.
1. PIXI.Text renders text with a 2d canvas. This is now only done once per unique text displayed in the graph.
2. PIXI.Graphics computes a mesh based on calls to its drawing methods. This is now only done once per unique node type in the graph.
3. PIXI.Text and PIXI.Graphics upload the rendered texture and mesh to webgl. Reusing the texture through PIXI.Sprite and mesh through PIXI.GraphicsGeometry means this is now done once.
4. Ports that are connected to another node are represented as their PIXI object and don't require the background graphics to be recomputed, instead they are rendered on top of the unfilled ports rendered in the background.
5. AudioGraphRender determines when nodes and edges are not in the given graph context and should be removed from its presentation. This is changed to reduce memory allocation (by creating strings) and reduce time spent doing this comparison.

### CODE REVIEW

File: src/panel/graph/AudioGraphRender.ts [hunk 1]
```diff
@@ -77,22 +85,29 @@ export class AudioGraphRender {
...
+    const {pixiApplication: app} = this;
 
     this.camera.setScreenSize(app.screen.width, app.screen.height);
     app.render();
   }
 
   /** Start rendering regularly. */
   start() {
-    this.renderFrameId = requestAnimationFrame(this.render);
+    // this.renderFrameId = requestAnimationFrame(this.render);
```
Comment (hoch) [inline src/panel/graph/AudioGraphRender.ts:110]: This line is commented. Is this intentional?

File: src/panel/graph/AudioNodeRender.ts [hunk 2]
```diff
@@ -235,36 +295,61 @@ export class AudioNodeRender {
...
+        : this.backgroundCacheGroup.plain
+    )
+      .getBackground(this.node)
+      .createMesh();
+
+    this.container.removeChild(this.background);
+    this.container.addChildAt(newBackground, 0);
+    this.background = newBackground;
+
+    // const {background, node} = this;
```
Comment (hoch) [inline src/panel/graph/AudioNodeRender.ts:327]: L327~L353 are commented.

File: src/panel/graph/AudioGraphRender.ts (hunk 1 above)
Comment (mzgoddard) [reply src/panel/graph/AudioGraphRender.ts]: It is intentional. Additionally I meant to remove `start`. It is replaced by `requestRender` in this changeset. `requestRender` is called when something has changed and the canvas rendering should be updated. This way the canvas only renders when the graph changes or the user pans or zooms the canvas.

**Review by hoch — APPROVED** [review]

**Comment by hoch** [comment]
@mzgoddard Do you want me to merge this to main? I see a series of PRs are waiting in the list, but wasn't sure about your plan.

**Comment by mzgoddard** [comment]
> @mzgoddard Do you want me to merge this to main? I see a series of PRs are waiting in the list, but wasn't sure about your plan.

Un. I'll go ahead and merge this and merge other PRs bottom up in the list that you've approved.

**Pull request merged**